lib/ansible/plugins/filter/core.py

```diff
@@ -283,11 +283,19 @@ def mandatory(a):
 
 def combine(*terms, **kwargs):
     recursive = kwargs.get('recursive', False)
-    if len(kwargs) > 1 or (len(kwargs) == 1 and 'recursive' not in kwargs):
-        raise AnsibleFilterError("'recursive' is the only valid keyword argument")
+    reverse = kwargs.get('reverse', False)
+
+    arg_error = "'recursive' and 'reverse' is the only valid keyword arguments"
+
+    for key in kwargs.keys():
+        if not (key == 'recursive' or key == 'reverse'):
+            raise AnsibleFilterError(arg_error)
+
+    if len(kwargs) > 2:
+        raise AnsibleFilterError(arg_error)
```

Comment:
This `if` isn't needed anymore. Because of the previous loop, there cannot be more than two elements in `kwargs` anyway.

Reply:
Indeed!